### ISSUE TITLE
Issue with parsed and clickable URLs being sent to the bot

### ISSUE THREAD

**Issue by xorist** [tomer8007/kik-bot-api-unofficial#138]
while testing a few things I noticed that whenever the bot receives some strange character that is after the "." in a parsed and clickable URL, (i.e., google.com) it doesn't know what to do with it and gives errors.
![Capture](https://user-images.githubusercontent.com/42919907/69768703-394e3b80-1147-11ea-8784-75e6b63d5b6a.PNG)
Above is an example, you can see the box there after the "." in the URL "Google.com".

**Comment by xorist** [comment]
After a bit of investigating I have discovered this character is possibly: U+200B : ZERO WIDTH SPACE [ZWSP]. Hope this helps.

**Comment by tomer8007** [comment]
What do you mean by having errors? What kind of errors does it show?

**Comment by ghost** [comment]
It doesn't properly parse nor respond to .com messages, if you were to put something like:

`
if chat_message.body == "hello.com":
`
and if someone said hello.com then it wouldn't act, but if you changed hello.com to hello in the if statement then it would be able to act. but during when hello.com exists, what would be sent to the terminal is:

`<body>google.\xe2\x80\x8bcom</body>
`
And
`
Exception in callback KikClient._on_new_data_received(b'<message to...er></message>')
handle: <Handle KikClient._on_new_data_received(b'<message to...er></message>')>
`
`
File "interactive_client.py", line 38, in on_group_message_received
    print("{} - {}: {}".format(friends[chat_message.group_jid].name, jid_to_username(chat_message.from_jid),
KeyError: '1100241318034_g@groups.kik.com'
`

**Comment by ghost** [comment]
this same traceback error wouldn't occur for echo_bot.py though, but echo_bot.py also wouldn't be able to act towards a '.com' text, or '.' at all

**Comment by sapiosaurus** [comment]
I don't see how this is relevant to the API at all, this is strictly Python related, not API.

As an aside, if you really want to extract URLs from a message, why not strip all non alphanumeric+symbol (or just strip non-ASCII) from the message and search for URLs with a regex pattern?

**Comment by xorist** [comment]
@drmarvel that's what I ended up doing. This is related to the API because if you recreate this issue you'll notice when echobot attempts to echo the link, it is not clickable.

**Comment by sapiosaurus** [comment]
but echo_bot is an example, not the API. If you're modifying an example for a non API function, you're outside of the scope of an API anyway.

**Comment by ghost** [comment]
@drmarvel i don't think this is python related as python is capable of parsing and accepting .com text with no issues.

**Comment by ghost** [comment]
echo_bot.py is an example of the API but it's also relying on the API. majority of the issues we run into are also related to the core API files.

**Comment by sapiosaurus** [comment]
right, but this is just text. There's nothing special about text containing a URL, or .com as it were. It's entirely **only** text and parsing. Python is not capable of detecting a URL with non printable characters in it without first stripping them out to detect it & action it.

**Comment by ghost** [comment]
But your API lacks the ability to strip these characters without running into issues

**Comment by sapiosaurus** [comment]
Of course it lacks that function, it just relays the text back to your own function. It's up to your function to do whatever it wants with the text. Data in - data out, that's all.

**Comment by ghost** [comment]
I suggest you guys implement the stripping capability, it would only make things better overall. Not all people are python geeks.

**Comment by xorist** [comment]
@drmarvel I agree echobot is an example, but even if I simply use only the functions provided with the API, the parsing for URLs does not withstand being copied from a user and being sent back. Since the character appears to be intended to indicate to the official client that it needs to be parsed, I think it makes sense that handling it should be a necessary aspect of the unofficial API.

**Comment by sapiosaurus** [comment]
```
import re
strippedMessage = re.sub(r'[^\x00-\x7f]',r'', chat_message.body)
```

**Comment by xorist** [comment]
@drmarvel we implemented a similar fix to this issue as you commented above, but it was something we had to figure out since the API was unable to relay the link correctly

**Comment by tomer8007** [comment]
I think the real question here is why this character is there. Just filtering out all non-printable characters is a workaround, not a solution. 

Since it seems to be the [Zero-width space](https://en.wikipedia.org/wiki/Zero-width_space) that basiclly just marks the end of the word, the real solution would be to parse the chat `body`  as a UTF-8 `unicode` instead of an `str`.

**Comment by xorist** [comment]
I think that's reasonable. This is the exact thing we ended up doing. Would you like me to make this edit?

**Comment by tomer8007** [comment]
Well, I think that the most elegant fix is to chnage [line #487 in client.py](https://github.com/tomer8007/kik-bot-api-unofficial/blob/e6db3c0e8a2f30473e4dc6ad8ac1d0bb45f01992/kik_unofficial/client.py#L487) to:

`xml_element = BeautifulSoup(data.decode('utf-8'), features='xml')`

What were you all ended up doing? If it's the same, I'll commit that.

**Comment by tomer8007** [comment]
Should be fixed in [this commit](https://github.com/tomer8007/kik-bot-api-unofficial/commit/5a2c290a3c291fd72a2d90ea321dfc919ac0ee33).
